src/main/java/org/apache/datasketches/theta/CompactSketch.java

```diff
@@ -19,25 +19,241 @@
+ * getResult() of a SetOperation, or from a heapify method.
  *
  * <p>A CompactSketch is the simplest form of a Theta Sketch. It consists of a compact list
  * (i.e., no intervening spaces) of hash values, which may be ordered or not, a value for theta
- * and a seed hash.  A CompactSketch is read-only,
+ * and a seed hash.  A CompactSketch is immutable (read-only),
  * and the space required when stored is only the space required for the hash values and 8 to 24
  * bytes of preamble. An empty CompactSketch consumes only 8 bytes.</p>
  *
  * @author Lee Rhodes
  */
 public abstract class CompactSketch extends Sketch {
+  private static final short defaultSeedHash = Util.computeSeedHash(DEFAULT_UPDATE_SEED);
+
+  /**
+   * Heapify takes a CompactSketch image in Memory and instantiates an on-heap CompactSketch.
+   *
+   * <p>The resulting sketch will not retain any link to the source Memory.</p>
+   *
+   * <p>This method assumes that the sketch image was created with the correct hash seed,
+   * so it is not checked.</p>
+   *
+   * @param srcMem an image of a CompactSketch.
+   * <a href="{@docRoot}/resources/dictionary.html#mem">See Memory</a>.
+   * @return a CompactSketch on the heap.
+   */
+  public static CompactSketch heapify(final Memory srcMem) {
+    final int serVer = srcMem.getByte(SER_VER_BYTE) & 0XFF;
+    final int familyID = srcMem.getByte(FAMILY_BYTE) & 0XFF;
+    final Family family = Family.idToFamily(familyID);
+    if (family != Family.COMPACT) {
+      throw new IllegalArgumentException("Corrupted: " + family + " is not Compact!");
+    }
+    if (serVer == 3) { //no seed check
+      final int flags = PreambleUtil.extractFlags(srcMem);
+      final boolean srcOrdered = (flags & ORDERED_FLAG_MASK) != 0;
+      return CompactOperations.memoryToCompact(srcMem, srcOrdered, null);
+    }
+    //not SerVer 3, assume compact stored form
+    if (serVer == 1) {
+      return ForwardCompatibility.heapify1to3(srcMem, defaultSeedHash);
+    }
+    if (serVer == 2) {
+      final short srcSeedHash = (short) extractSeedHash(srcMem);
+      return ForwardCompatibility.heapify2to3(srcMem, srcSeedHash);
+    }
+    throw new SketchesArgumentException("Unknown Serialization Version: " + serVer);
+  }
+
+  /**
+   * Heapify takes a CompactSketch image in Memory and instantiates an on-heap CompactSketch.
+   *
+   * <p>The resulting sketch will not retain any link to the source Memory.</p>
+   *
+   * <p>This this method checks if the given seed was used to create the source Memory image.
+   * However, SerialVersion 1 sketches cannot be checked.</p>
+   *
+   * @param srcMem an image of a CompactSketch that was created using the given seed.
+   * <a href="{@docRoot}/resources/dictionary.html#mem">See Memory</a>.
+   * @param seed <a href="{@docRoot}/resources/dictionary.html#seed">See Update Hash Seed</a>.
+   * @return a CompactSketch on the heap.
+   */
+  public static CompactSketch heapify(final Memory srcMem, final long seed) {
```

Comment:
we should think about renaming `seed` here to `expectedSeed`, as mostly seed is being used here as a validation of the seedHash that is in the `srcMem`, and not setting the seed to be a specific value (v1 sketches excepted). Even though I know what this was supposed to do, I got confused reading the code because I was thinking this was passing in the seed  rather than specifying the expected value to validate against.

Reply:
Same comment as above.